### ISSUE TITLE
Linked property and loan not working properly

### ISSUE THREAD

**Issue by BrendanJM** [mintapi/mintapi#197]
In mint, you have the ability to link a property and loans on the accounts page via the dropdown: "Is this property associated with a loan, mortgage, or line of credit?". 

However, when pulling the data for either a property or a loan, there doesn't appear to be anywhere that indicates any sort of linkage. There are fields that denote account linkage (perhaps this is something different?), but these do not appear to populate for me.

Having this functionality would be great for equity calculation.

**Comment by mrooney** [comment]
Thanks for the info. This is just pulling from the REST API, so it may not be available in the response we obtain.

**Comment by burkematthew** [comment]
@mrooney @jaron-l I found a way to get this information from Mint, but it's in a completely different endpoint than what we're using for accounts.  However, I actually think switching to this new endpoint, `providers`, could be a positive change for a few different reasons (which we can discuss elsewhere as it relates to the next major release as it would be a breaking change).  

Also, on a slightly different note, I haven't been able to even trace a call to `bundledServiceController.xevent?legacy=false` for account-level data to even get a feel for how we might be able to change the request to get the linked data.  It seems possible that Mint deprecated their use of `bundledServiceController.xevent?legacy=false` within their application for the `providers` endpoint, but didn't remove the functionality that we depend on, which is why it still works.  

In a nutshell, I think that to fulfill this request, we'll need to move to the new endpoint.

**Comment by mrooney** [comment]
> @mrooney @jaron-l I found a way to get this information from Mint, but it's in a completely different endpoint than what we're using for accounts. However, I actually think switching to this new endpoint, `providers`, could be a positive change for a few different reasons (which we can discuss elsewhere as it relates to the next major release as it would be a breaking change).
> 
> Also, on a slightly different note, I haven't been able to even trace a call to `bundledServiceController.xevent?legacy=false` for account-level data to even get a feel for how we might be able to change the request to get the linked data. It seems possible that Mint deprecated their use of `bundledServiceController.xevent?legacy=false` within their application for the `providers` endpoint, but didn't remove the functionality that we depend on, which is why it still works.
> 
> In a nutshell, I think that to fulfill this request, we'll need to move to the new endpoint.

I think that makes sense and I'm in favor of that. I am imagining a world where they haven't removed it yet because they are still seeing hits to it for reasons they don't understand, and those are from us 🤣

**Comment by burkematthew** [comment]
@mrooney at what point do we want to move to a new major release?  Should I hold off on working on moving to the new `providers` endpoint or am I good to start working?  If it'll be awhile before we actually want to deploy it, I'll hold off and work on other non-breaking changes.

**Comment by jaron-l** [comment]
Just a thought, can this be implemented in a backwards compatible manner? Can you just add a new method and show a deprecation?

**Comment by burkematthew** [comment]
We can certainly implement a new method for the `providers` endpoint but it won't link up to the command-line, which is I guess a good short term solution to at least get the endpoint implemented and people used to seeing the new data structure.

**Comment by mrooney** [comment]
> @mrooney at what point do we want to move to a new major release? Should I hold off on working on moving to the new `providers` endpoint or am I good to start working? If it'll be awhile before we actually want to deploy it, I'll hold off and work on other non-breaking changes.

Is it a pretty large code change, or a smaller code change and just using a new endpoint under the hood? If the latter, I'm fine with just doing it as a point release. If it is a larger code change, you raise a good point and we can do it as a v2.0 release after wrapping up any smaller in-flight PRs :)

**Comment by burkematthew** [comment]
@mrooney the change itself is not a big deal to make. However, the differences lie in the underlying data returned: different fields, different labels to describe similar data, etc. It's different enough that it'd be a breaking change for retrieving account data.

**Comment by mrooney** [comment]
@burkematthew Got it! In that case, I like the idea of releasing that as a v2 just for extra clarity, and we can add plenty of notes :)

**Comment by burkematthew** [comment]
@mrooney so hold off for a little bit on this until we're ready for v2?

**Comment by mrooney** [comment]
I'm happy to see it implemented either as a new method/switch that doesn't break anything, or release the new endpoint usage as a v2. Also, is it pretty different or would it make sense to simply map the data to the existing format?

**Comment by burkematthew** [comment]
@mrooney in terms of effort to go through mapping to the existing format to just break it on v2, I'd say it's not worth it.  I'd rather wait or at least implement a new method that doesn't break anything.

**Comment by burkematthew** [comment]
@BrendanJM can you pull the latest from main and see if this is resolved? Last I checked, the linked property and accounts seemed to be working with the new endpoint.

**Comment by BrendanJM** [comment]
I'm not really using mint or mintapi at this time, so I might not be the best person to confirm the fix. If it appears resolved to you, feel free to close this out.
